### ISSUE TITLE
Error: group name with '&', or sth. else with '&'

### ISSUE THREAD

**Issue by mysteriouss** [blueset/ehForwarderBot#17]
```python
2017-01-21 11:23:00,758: telegram.ext.updater [ERROR]
    Error while getting Updates: Invalid server response
2017-01-21 11:23:00,759: telegram.ext.updater [ERROR]
    unhandled exception
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/site-packages/telegram/ext/updater.py", line 115, in _thread_wrapper
    target(*args, **kwargs)
  File "/usr/local/lib/python3.5/site-packages/telegram/ext/dispatcher.py", line 219, in start
    self.process_update(update)
  File "/usr/local/lib/python3.5/site-packages/telegram/ext/dispatcher.py", line 263, in process_update
    self.dispatch_error(None, update)
  File "/usr/local/lib/python3.5/site-packages/telegram/ext/dispatcher.py", line 374, in dispatch_error
    callback(self.bot, update, error)
  File "/root/ehForwarderBot/plugins/eh_telegram_master/__init__.py", line 1186, in error
    (html.escape(error), html.escape(update)))
  File "/usr/local/lib/python3.5/html/__init__.py", line 19, in escape
    s = s.replace("&", "&amp;") # Must be done first!
AttributeError: 'TelegramError' object has no attribute 'replace'
Exception in thread dispatcher:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/threading.py", line 923, in _bootstrap_inner
    self.run()
  File "/usr/local/lib/python3.5/threading.py", line 871, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.5/site-packages/telegram/ext/updater.py", line 115, in _thread_wrapper
    target(*args, **kwargs)
  File "/usr/local/lib/python3.5/site-packages/telegram/ext/dispatcher.py", line 219, in start
    self.process_update(update)
  File "/usr/local/lib/python3.5/site-packages/telegram/ext/dispatcher.py", line 263, in process_update
    self.dispatch_error(None, update)
  File "/usr/local/lib/python3.5/site-packages/telegram/ext/dispatcher.py", line 374, in dispatch_error
    callback(self.bot, update, error)
  File "/root/ehForwarderBot/plugins/eh_telegram_master/__init__.py", line 1186, in error
    (html.escape(error), html.escape(update)))
  File "/usr/local/lib/python3.5/html/__init__.py", line 19, in escape
    s = s.replace("&", "&amp;") # Must be done first!
```

**Comment by mysteriouss** [comment]
```python 

File "/usr/local/lib/python3.5/site-packages/telegram/ext/dispatcher.py", line 270, in process_update
    handler.handle_update(update, self)
  File "/usr/local/lib/python3.5/site-packages/telegram/ext/regexhandler.py", line 105, in handle_update
    return self.callback(dispatcher.bot, update, **optional_args)
  File "/root/ehForwarderBot/plugins/eh_telegram_master/__init__.py", line 803, in extra_call
    bot.editMessageText(text=header + result, chat_id=update.message.chat.id, message_id=msg.message_id)
  File "/usr/local/lib/python3.5/site-packages/telegram/bot.py", line 125, in decorator
    result = func(self, *args, **kwargs)
  File "/usr/local/lib/python3.5/site-packages/telegram/bot.py", line 151, in decorator
    result = self._request.post(url, data, timeout=kwargs.get('timeout'))
  File "/usr/local/lib/python3.5/site-packages/telegram/utils/request.py", line 207, in post
    **urlopen_kwargs)
  File "/usr/local/lib/python3.5/site-packages/telegram/utils/request.py", line 152, in _request_wrapper
    raise BadRequest(repr(message))
telegram.error.BadRequest: 'Bad Request: MESSAGE_TOO_LONG'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.5/site-packages/telegram/ext/dispatcher.py", line 278, in process_update
    self.dispatch_error(update, te)
  File "/usr/local/lib/python3.5/site-packages/telegram/ext/dispatcher.py", line 374, in dispatch_error
    callback(self.bot, update, error)
  File "/root/ehForwarderBot/plugins/eh_telegram_master/__init__.py", line 1186, in error
    (html.escape(error), html.escape(update)))
  File "/usr/local/lib/python3.5/html/__init__.py", line 19, in escape
    s = s.replace("&", "&amp;") # Must be done first!
AttributeError: 'BadRequest' object has no attribute 'replace'
2017-01-22 10:00:34,837: telegram.ext.dispatcher [ERROR]
    An uncaught error was raised while handling the error
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/site-packages/telegram/ext/dispatcher.py", line 270, in process_update
    handler.handle_update(update, self)
  File "/usr/local/lib/python3.5/site-packages/telegram/ext/regexhandler.py", line 105, in handle_update
    return self.callback(dispatcher.bot, update, **optional_args)
  File "/root/ehForwarderBot/plugins/eh_telegram_master/__init__.py", line 803, in extra_call
    bot.editMessageText(text=header + result, chat_id=update.message.chat.id, message_id=msg.message_id)
  File "/usr/local/lib/python3.5/site-packages/telegram/bot.py", line 125, in decorator
    result = func(self, *args, **kwargs)
  File "/usr/local/lib/python3.5/site-packages/telegram/bot.py", line 151, in decorator
    result = self._request.post(url, data, timeout=kwargs.get('timeout'))
  File "/usr/local/lib/python3.5/site-packages/telegram/utils/request.py", line 207, in post
    **urlopen_kwargs)
  File "/usr/local/lib/python3.5/site-packages/telegram/utils/request.py", line 152, in _request_wrapper
    raise BadRequest(repr(message))
telegram.error.BadRequest: 'Bad Request: MESSAGE_TOO_LONG'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.5/site-packages/telegram/ext/dispatcher.py", line 278, in process_update
    self.dispatch_error(update, te)
  File "/usr/local/lib/python3.5/site-packages/telegram/ext/dispatcher.py", line 374, in dispatch_error
    callback(self.bot, update, error)
  File "/root/ehForwarderBot/plugins/eh_telegram_master/__init__.py", line 1186, in error
    (html.escape(error), html.escape(update)))
  File "/usr/local/lib/python3.5/html/__init__.py", line 19, in escape
    s = s.replace("&", "&amp;") # Must be done first!
AttributeError: 'BadRequest' object has no attribute 'replace'
```

**Comment by blueset** [comment]
Trying to address it in c941407. As no specific reproduction steps is provided, please try if it's fixed in the latest dev commit, or wait it to be merged to the stable branch.

**Comment by mysteriouss** [comment]
ok, thanks

**Comment by mysteriouss** [comment]
```python
2017-01-23 10:33:33,347: telegram.ext.updater [ERROR]
    Error while getting Updates: Timed out
2017-01-23 10:33:43,351: telegram.ext.updater [ERROR]
    unhandled exception
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/site-packages/urllib3/connectionpool.py", line 600, in urlopen
    chunked=chunked)
  File "/usr/local/lib/python3.5/site-packages/urllib3/connectionpool.py", line 345, in _make_request
    self._validate_conn(conn)
  File "/usr/local/lib/python3.5/site-packages/urllib3/connectionpool.py", line 844, in _validate_conn
    conn.connect()
  File "/usr/local/lib/python3.5/site-packages/urllib3/connection.py", line 326, in connect
    ssl_context=context)
  File "/usr/local/lib/python3.5/site-packages/urllib3/util/ssl_.py", line 324, in ssl_wrap_socket
    return context.wrap_socket(sock, server_hostname=server_hostname)
  File "/usr/local/lib/python3.5/ssl.py", line 376, in wrap_socket
    _context=self)
  File "/usr/local/lib/python3.5/ssl.py", line 747, in __init__
    self.do_handshake()
  File "/usr/local/lib/python3.5/ssl.py", line 983, in do_handshake
    self._sslobj.do_handshake()
  File "/usr/local/lib/python3.5/ssl.py", line 628, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLEOFError: EOF occurred in violation of protocol (_ssl.c:646)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.5/site-packages/telegram/utils/request.py", line 132, in _request_wrapper
    resp = self._con_pool.request(*args, **kwargs)
  File "/usr/local/lib/python3.5/site-packages/urllib3/request.py", line 70, in request
    **urlopen_kw)
  File "/usr/local/lib/python3.5/site-packages/urllib3/request.py", line 148, in request_encode_body
    return self.urlopen(method, url, **extra_kw)
  File "/usr/local/lib/python3.5/site-packages/urllib3/poolmanager.py", line 244, in urlopen
    response = conn.urlopen(method, u.request_uri, **kw)
  File "/usr/local/lib/python3.5/site-packages/urllib3/connectionpool.py", line 630, in urlopen
    raise SSLError(e)
urllib3.exceptions.SSLError: EOF occurred in violation of protocol (_ssl.c:646)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.5/site-packages/telegram/ext/updater.py", line 115, in _thread_wrapper
    target(*args, **kwargs)
  File "/usr/local/lib/python3.5/site-packages/telegram/ext/dispatcher.py", line 219, in start
    self.process_update(update)
  File "/usr/local/lib/python3.5/site-packages/telegram/ext/dispatcher.py", line 263, in process_update
    self.dispatch_error(None, update)
  File "/usr/local/lib/python3.5/site-packages/telegram/ext/dispatcher.py", line 374, in dispatch_error
    callback(self.bot, update, error)
  File "/root/ehForwarderBot/plugins/eh_telegram_master/__init__.py", line 1196, in error
    (html.escape(str(error)), html.escape(str(update))))
  File "/usr/local/lib/python3.5/site-packages/telegram/bot.py", line 125, in decorator
    result = func(self, *args, **kwargs)
  File "/usr/local/lib/python3.5/site-packages/telegram/bot.py", line 151, in decorator
    result = self._request.post(url, data, timeout=kwargs.get('timeout'))
  File "/usr/local/lib/python3.5/site-packages/telegram/utils/request.py", line 207, in post
    **urlopen_kwargs)
  File "/usr/local/lib/python3.5/site-packages/telegram/utils/request.py", line 138, in _request_wrapper
    raise NetworkError('urllib3 HTTPError {0}'.format(error))
telegram.error.NetworkError: urllib3 HTTPError EOF occurred in violation of protocol (_ssl.c:646)
Exception in thread dispatcher:
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/site-packages/urllib3/connectionpool.py", line 600, in urlopen
    chunked=chunked)
  File "/usr/local/lib/python3.5/site-packages/urllib3/connectionpool.py", line 345, in _make_request
    self._validate_conn(conn)
  File "/usr/local/lib/python3.5/site-packages/urllib3/connectionpool.py", line 844, in _validate_conn
    conn.connect()
  File "/usr/local/lib/python3.5/site-packages/urllib3/connection.py", line 326, in connect
    ssl_context=context)
  File "/usr/local/lib/python3.5/site-packages/urllib3/util/ssl_.py", line 324, in ssl_wrap_socket
    return context.wrap_socket(sock, server_hostname=server_hostname)
  File "/usr/local/lib/python3.5/ssl.py", line 376, in wrap_socket
    _context=self)
  File "/usr/local/lib/python3.5/ssl.py", line 747, in __init__
    self.do_handshake()
  File "/usr/local/lib/python3.5/ssl.py", line 983, in do_handshake
    self._sslobj.do_handshake()
  File "/usr/local/lib/python3.5/ssl.py", line 628, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLEOFError: EOF occurred in violation of protocol (_ssl.c:646)


During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.5/threading.py", line 923, in _bootstrap_inner
    self.run()
  File "/usr/local/lib/python3.5/threading.py", line 871, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/local/lib/python3.5/site-packages/telegram/ext/updater.py", line 115, in _thread_wrapper
    target(*args, **kwargs)
  File "/usr/local/lib/python3.5/site-packages/telegram/ext/dispatcher.py", line 219, in start
    self.process_update(update)
  File "/usr/local/lib/python3.5/site-packages/telegram/ext/dispatcher.py", line 263, in process_update
    self.dispatch_error(None, update)
  File "/usr/local/lib/python3.5/site-packages/telegram/ext/dispatcher.py", line 374, in dispatch_error
    callback(self.bot, update, error)
  File "/root/ehForwarderBot/plugins/eh_telegram_master/__init__.py", line 1196, in error
    (html.escape(str(error)), html.escape(str(update))))
  File "/usr/local/lib/python3.5/site-packages/telegram/bot.py", line 125, in decorator
    result = func(self, *args, **kwargs)
  File "/usr/local/lib/python3.5/site-packages/telegram/bot.py", line 151, in decorator
    result = self._request.post(url, data, timeout=kwargs.get('timeout'))
  File "/usr/local/lib/python3.5/site-packages/telegram/utils/request.py", line 207, in post
    **urlopen_kwargs)
  File "/usr/local/lib/python3.5/site-packages/telegram/utils/request.py", line 138, in _request_wrapper
    raise NetworkError('urllib3 HTTPError {0}'.format(error))
telegram.error.NetworkError: urllib3 HTTPError EOF occurred in violation of protocol (_ssl.c:646)

2017-01-23 10:33:44,353: telegram.ext.updater [ERROR]
    Error while getting Updates: urllib3 HTTPError EOF occurred in violation of protocol (_ssl.c:646)
2017-01-23 10:33:55,858: telegram.ext.updater [ERROR]
    Error while getting Updates: urllib3 HTTPError EOF occurred in violation of protocol (_ssl.c:646)
2017-01-23 10:34:18,152: telegram.ext.updater [ERROR]
    Error while getting Updates: Timed out
```

^C[$ ehForwarderBot]# ^C
[root@centos ehForwarderBot]# python35 /root/ehForwarderBot/daemon.py restart
Restarting EFB daemon (1)

**Comment by blueset** [comment]
My suggestion is to switch to a more stable internet connection. Frequent timeout while connecting to Telegram Bot API is an extremely rare case. Please also ensure that your server is not hosted in where Telegram is potentially blocked (e.g. mainland China, etc.).

**Comment by mysteriouss** [comment]
ok,it's ashamed that I'm using a shadowsocks-proxied  server at home in mainland China.

**Comment by blueset** [comment]
It is fine to test EFB locally behind any proxy, as long as it's applicable. However, it's not recommended to do so in a production environment, since the proxy is not always stable, and it may cause unexpected problems. AFAIK, my EFB hosted in Japan ([Vultr JP](http://www.vultr.com/?ref=6951220-3B)) works prefectly with both WeChat and Telegram.

**Comment by mysteriouss** [comment]
Cannot send but able to receive.

**Comment by blueset** [comment]
Please open another issue, and report properly with log.
